### PR TITLE
Preprocessing text and automatic spell checking

### PR DESCRIPTION
**Pull request by dgarrett622**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
see #55 

##### What are the significant changes in functionality due to this change request?
Adds capability to preprocess text using textacy and numerizer. Adds automatic spelling correction. Adds Jupyter notebooks to demonstrate preprocessing capabilities.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual.
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Regression tests have to complete successfully.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test.
- [x] 6. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.

### CODE REVIEW

File: src/nlp/Preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+import re
+import textacy.preprocessing as preprocessing
+from numerizer import numerize
+import spacy
+from spacy.vocab import Vocab
+from contextualSpellCheck.contextualSpellCheck import ContextualSpellCheck
+import autocorrect
+
+# list of available preprocessors in textacy.preprocessing.normalize
+textacy_normalize = ['bullet_points',
```
Comment (wangcj05) [inline src/nlp/Preprocessing.py]: we are following RAVEN code standards, could you change it to camelBack?

File: src/nlp/Preprocessing.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                     'hyphenated_words',
+                     'quotation_marks',
+                     'repeating_chars',
+                     'unicode',
+                     'whitespace']
+# list of available preprocessors in textacy.preprocessing.remove
+textacy_remove = ['accents',
```
Comment (wangcj05) [inline src/nlp/Preprocessing.py]: camelBack here?

File: src/nlp/Preprocessing.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+                  'brackets',
+                  'html_tags',
+                  'punctuation']
+# list of available preprocessors in textacy.preprocessing.replace
+textacy_replace = ['currency_symbols',
```
Comment (wangcj05) [inline src/nlp/Preprocessing.py]: camelBack here?

**Review by wangcj05 — CHANGES_REQUESTED** [review]
@dgarrett622 I have some general comments for you to consider.

File: src/nlp/Preprocessing.py (hunk 1 above)
Comment (dgarrett622) [reply src/nlp/Preprocessing.py]: changed to textacyNormalize

File: src/nlp/Preprocessing.py (hunk 2 above)
Comment (dgarrett622) [reply src/nlp/Preprocessing.py]: changed to textacyRemove

File: src/nlp/Preprocessing.py (hunk 3 above)
Comment (dgarrett622) [reply src/nlp/Preprocessing.py]: changed to textacyReplace

**Review by wangcj05 — APPROVED** [review]

**Comment by wangcj05** [comment]
PR is good, and checklist is satisfied.

**Pull request merged**